js/src/workspaces/window.js

```diff
@@ -762,22 +822,28 @@
 
     setCurrentCanvasID: function(canvasID) {
       var _this = this;
-      this.canvasID = canvasID;
+
+      _this.canvasID = canvasID;
+
+      // TODO: Completely remove the 'removeTooltips' event. They are not actions and
+      // they do not represent any state change. They are essentially
+      // function calls, which should be handled as responses to the
+      // publication of the state update event currentCanvasIDUpdated
+      // below.
       _this.eventEmitter.publish('removeTooltips.' + _this.id);
-      _this.eventEmitter.unsubscribe(('annotationListLoaded.' + _this.id));
```

Comment:
Because the annotationsLayer is only being initialised once for the window, it will not re-subscribe itself, so this is no longer required. In the initialisation sequence, this event is actually called _after_ the subscription is bound for the first time, so this was causing the event to never be called.